### ISSUE TITLE
fetch Segmentation Fault

### ISSUE THREAD

**Issue by TrackZone1** [oven-sh/bun#6415]
### What version of Bun is running?

1.0.5+1bf28e0d77a8b2261befbdb708cefd03e0126960

### What platform is your computer?

Linux 6.2.16-12-pve x86_64 unknown

### What steps can reproduce the bug?

```
Bun.serve({
    port: 80,
    async fetch(req, srv) {
        const url = req.url;
        const urlProxies = url.replace("#ipReverse#", "#ipTarget#");

        const res = await fetch(urlProxies, req);
        return res;
    },
});
```

**Comment by Electroid** [comment]
Hi, thanks for reporting this issue. There isn't enough information in that code snippet to reproduce this issue. Can you paste the full code?

**Comment by TrackZone1** [comment]
this is the complete code I forgot to mention it was with 500 simultaneous connections

**Comment by liz3** [comment]
This code would create a infinite request loop, on my machine it does not crash but gets stuck.
Im not entirely sure though what the expected behaviour should be?
Heres the code i tested with
```js
Bun.serve({
    port: 20132,
    async fetch(req) {
        console.log(req.url)
        const res = await fetch(req.url, req)
        return res;
    },
})
```
## Edit
I went over your code again and it might be the case that you are redirecting to another host which is a ip address?
In that case it might be useful to have a bit more context to reproduce this issue

**Comment by TrackZone1** [comment]
The principle of a reverse proxy is to make a request to another URL to retrieve its content, what you do is copy the url of your server which is already the one whose mission is to display the content so you contact your same server with each request

```
Bun.serve({
    port: 80,
    async fetch(req, srv) {
        const url = req.url;

        // #ipReverse is 1.3.3.7 and  #ipTarget is 7.3.3.1
        const urlProxies = url.replace("#ipReverse#", "#ipTarget#");

        const res = await fetch(urlProxies, req);
        return res;
    },
});
```

![image](https://github.com/oven-sh/bun/assets/48413426/a0856767-aed8-43e1-bdcf-6479f095bed3)

**Comment by liz3** [comment]
> The principle of a reverse proxy is to make a request to another URL to retrieve its content, what you do is copy the url of your server which is already the one whose mission is to display the content so you contact your same server with each request
> 
> 
> 
> ```
> 
> Bun.serve({
> 
>     port: 80,
> 
>     async fetch(req, srv) {
> 
>         const url = req.url;
> 
> 
> 
>         // #ipReverse is 1.3.3.7 and  #ipTarget is 7.3.3.1
> 
>         const urlProxies = url.replace("#ipReverse#", "#ipTarget#");
> 
> 
> 
>         const res = await fetch(urlProxies, req);
> 
>         return res;
> 
>     },
> 
> });
> 
> ```
> 
> 
> 
> ![image](https://github.com/oven-sh/bun/assets/48413426/943e3369-196b-4387-896c-2c0c7ede9e59)
> 
> 

Yes im aware, i still think it might be useful to have some more info to reproduce the issue, the fact that you are doing a reverse proxy in aware of.

**Comment by TrackZone1** [comment]
https://github.com/oven-sh/bun/issues/6415#issuecomment-1758310772
